### PR TITLE
Feature/cvsb 7677

### PR DESCRIPTION
**Pull request by m-ujaffer**

### CODE REVIEW

File: src/services/CertificateGenerationService.ts [hunk 1]
```diff
@@ -62,10 +46,16 @@ class CertificateGenerationService {
...
+            trl_prs: config.documentNames.trl_prs,
+            rwt: config.documentNames.rwt
         };
 
-        const vehicleTestRes: string = testResult.vehicleType + "_" + testType.testResult;
+        let vehicleTestRes: string;
+        if (CertificateGenerationService.isRoadworthinessTestType(testType.testTypeId)) { // CVSB-7677 is roadworthisness test
+            vehicleTestRes = "rwt";
+        } else  {
+        vehicleTestRes = testResult.vehicleType + "_" + testType.testResult;
```
Comment (KHodderDVSA) [inline src/services/CertificateGenerationService.ts]: linting should have complained about tabulation

Comment (m-ujaffer) [reply src/services/CertificateGenerationService.ts]: verified with npm run tslint. Its not throwing an errors.

File: src/services/CertificateGenerationService.ts [hunk 2]
```diff
@@ -62,10 +46,16 @@ class CertificateGenerationService {
...
             trl_pass: config.documentNames.vtg5a,
             trl_fail: config.documentNames.vtg30,
-            trl_prs: config.documentNames.trl_prs
+            trl_prs: config.documentNames.trl_prs,
+            rwt: config.documentNames.rwt
         };
 
-        const vehicleTestRes: string = testResult.vehicleType + "_" + testType.testResult;
+        let vehicleTestRes: string;
+        if (CertificateGenerationService.isRoadworthinessTestType(testType.testTypeId)) { // CVSB-7677 is roadworthisness test
```
Comment (KHodderDVSA) [inline src/services/CertificateGenerationService.ts]: image
No test for this

File: src/services/CertificateGenerationService.ts [hunk 3]
```diff
@@ -133,63 +121,139 @@ class CertificateGenerationService {
...
+                };
+                return resultPass;
+        }
     }
 
+    /**
+     * Retrieves the vehicle weight details for Roadworthisness certificates
+     * @param testResult
+     */
+    public async getWeightDetails(testResult: any) {
```
Comment (KHodderDVSA) [inline src/services/CertificateGenerationService.ts]: Look over missing test cases - leave it up to you what is important
image

File: src/services/CertificateGenerationService.ts [hunk 4]
```diff
@@ -252,11 +316,31 @@ class CertificateGenerationService {
...
 
     /**
-     * Retrieves the vehicle make and model for a given vehicle from the Technical Records microservice.
-     * Dramatically altered in CVSB-8582, in order to manage failure of vin data to conform to agreed formats
-     * @param testResult - the full test result record, from which vehicle attributes to search on can be obtained
+     * Method for getting make and model based on the vehicle from a test-result
+     * @param testResult - the testResult for which the tech record search is done for
      */
     public async getVehicleMakeAndModel(testResult: any) {
+        const techRecord = await this.getTechRecord(testResult);
```
Comment (KHodderDVSA) [inline src/services/CertificateGenerationService.ts]: I know this wasn't really your change, but could this get the full coverage?
image

**Pull request merged**